lib/src/native/realm_core.dart

```diff
@@ -47,9 +47,26 @@ import '../user.dart';
 import '../set.dart';
 import 'realm_bindings.dart';
 
-late RealmLibrary _realmLib;
+const bugInTheSdkMessage = "This is likely a bug in the Realm SDK - please file an issue at https://github.com/realm/realm-dart/issues";
 
-final _RealmCore realmCore = _RealmCore();
+final _realmLib = () {
```

Comment:
I would suggest we move decimal128.dart into the native directory and keeping the initialization of realmCore as is.
_realmLib is intentionally hidden from the rest of the SDK. We could declare deciam128 as a partial file to realm_core.dart if we want to keep it separate from the rest of the native invocations. Which I agree is better.

Reply:
I tried doing this, but it gets ugly quickly. Fx. we then need to `show` a class from `native/realm_core.dart`. I'm not sure we want to go that way. What do you think?

Reply:
You can see the changes in the [Move decimal128.dart to native as parts file of realm_core.dart](https://github.com/realm/realm-dart/pull/1192/commits/c9433333669cbf74c8298df66b136c2c9be2e2c0) commit

Reply:
I think its better. I like that we are not importing ffi and bindings stuff into non native part of the SDK.

The only "downside" is we need to do 
> export 'native/realm_core.dart' show Decimal128;

but that's not an issue since we are in control there.